pkg/cmd/server/startconfig/start_master.go

```diff
@@ -0,0 +1,245 @@
+Kubernetes cluster.
+`
+
+// NewCommandStartMaster provides a CLI handler for 'start' command
+func NewCommandStartMaster() *cobra.Command {
+	options := &MasterOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "master",
+		Short: "Launch OpenShift master",
+		Long:  longMasterCommandDesc,
+		Run: func(c *cobra.Command, args []string) {
+			if err := options.Complete(); err != nil {
+				glog.Fatal(err)
+			}
+
+			if err := options.Validate(); err != nil {
+				glog.Fatal(err)
+			}
+
+			if err := options.StartMaster(); err != nil {
+				glog.Fatal(err)
+			}
+		},
+	}
+
+	flags := cmd.Flags()
+
+	flags.BoolVar(&options.WriteConfigOnly, "config-only", false, "Indicates that the command should build the config that would be used to start OpenShift and do nothing else.")
+	flags.StringVar(&options.ConfigFile, "config", "", "Indicates that the command use the config file from a certain location.  If this is specified, all other options are ignored except --config-only.")
+
+	options.MasterArgs = NewDefaultMasterArgs()
+	// make sure that KubeConnectionArgs and MasterArgs use the same CertArgs for this command
+	options.MasterArgs.KubeConnectionArgs.CertArgs = options.MasterArgs.CertArgs
+
+	BindMasterArgs(options.MasterArgs, flags, "")
+	BindBindAddrArg(options.MasterArgs.BindAddrArg, flags, "")
+	BindImageFormatArgs(options.MasterArgs.ImageFormatArgs, flags, "")
+	BindKubeConnectionArgs(options.MasterArgs.KubeConnectionArgs, flags, "")
+	BindCertArgs(options.MasterArgs.CertArgs, flags, "")
+
+	return cmd
+}
+
+func (o MasterOptions) Validate() error {
+	return nil
+}
+
+func (o MasterOptions) Complete() error {
+	nodeList := util.NewStringSet()
+	// take everything toLower
+	for _, s := range o.MasterArgs.NodeList {
+		nodeList.Insert(strings.ToLower(s))
+	}
+
+	o.MasterArgs.NodeList = nodeList.List()
+
+	return nil
+}
+
+const (
+	unauthenticatedUsername = "system:anonymous"
+
+	authenticatedGroup   = "system:authenticated"
+	unauthenticatedGroup = "system:unauthenticated"
+)
+
+// run launches the appropriate startup modes or returns an error.
+func (o MasterOptions) StartMaster() error {
+	startUsingConfigFile := !o.WriteConfigOnly && (len(o.ConfigFile) > 0)
+	mintCerts := o.MasterArgs.CertArgs.MintCerts && !startUsingConfigFile
+
+	if mintCerts {
+		if err := o.MintCerts(); err != nil {
+			return nil
+		}
+	}
+
+	var masterConfig *configapi.OpenShiftMasterConfig
+	var err error
+	if startUsingConfigFile {
+		masterConfig, err = ReadMasterConfig(o.ConfigFile)
+	} else {
+		masterConfig, err = o.MasterArgs.BuildSerializeableOpenShiftMasterConfig()
+	}
+	if err != nil {
+		return err
+	}
+
+	if o.WriteConfigOnly {
+		content, err := WriteMaster(masterConfig)
+		if err != nil {
+			return err
+		}
+		if err := ioutil.WriteFile(o.ConfigFile, content, 0644); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	if err := start.StartMaster(masterConfig); err != nil {
+		return err
+	}
+
+	daemon.SdNotify("READY=1")
```

Comment:
We're sending this twice with an all-in-one start... I don't know what the implications of that are